File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/UpdateScheduleDao.java
package com.platdmit.simplecloudmanager.data.database.dao;

import com.platdmit.simplecloudmanager.data.database.entity.DbUpdateSchedule;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface UpdateScheduleDao extends BaseDao<DbUpdateSchedule>{

    @Query("SELECT * FROM dbupdateschedule")
    List<DbUpdateSchedule> getScheduleAll();

    @Query("SELECT * FROM dbupdateschedule WHERE mName = :name")
    DbUpdateSchedule getScheduleForName(String name);

    @Query("SELECT * FROM dbupdateschedule WHERE mNextUpdate <= :data")
    List<DbUpdateSchedule> getScheduleForUpdate(Long data);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/UpdateScheduleDao.kt
package com.platdmit.simplecloudmanager.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.platdmit.simplecloudmanager.data.database.entity.DbUpdateSchedule

@Dao
interface UpdateScheduleDao : BaseDao<DbUpdateSchedule> {
    @Query("SELECT * FROM dbupdateschedule")
    fun getScheduleAll(): List<DbUpdateSchedule>?

    @Query("SELECT * FROM dbupdateschedule WHERE name = :name")
    fun getScheduleForName(name: String): DbUpdateSchedule?

    @Query("SELECT * FROM dbupdateschedule WHERE nextUpdate <= :data")
    fun getScheduleForUpdate(data: Long): List<DbUpdateSchedule>?
}